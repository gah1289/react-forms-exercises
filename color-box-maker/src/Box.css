.delete-btn {
	position: relative;
	background-color: rgba(0, 0, 0, 0.5);
	color: rgba(255, 255, 255, 0.5);
	margin-right: 0;
	margin-left: auto;
	display: block;
	font-weight: bold;
	border: none;
}

.delete-btn:hover {
	background-color: rgba(0, 0, 0, 0.6);
	color: rgba(255, 255, 255, 1);
}

.hide-box {
	display: none;
}

.Box {
	margin: 10px;
	padding: 1px;
}
