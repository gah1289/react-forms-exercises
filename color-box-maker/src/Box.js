import React, { useState } from 'react';

import './Box.css';

const Box = ({ backgroundColor, width, height }) => {
	const [
		hideBox,
		setHideBox
	] = useState(false);

	return (
		<div
			className={` Box ${hideBox ? 'hide-box' : ''}`}
			style={{ backgroundColor, width: Number(width), height: Number(height) }}
		>
			{' '}
			<button onClick={() => setHideBox(true)} className={`delete-btn`}>
				x
			</button>
		</div>
	);
};

export default Box;
