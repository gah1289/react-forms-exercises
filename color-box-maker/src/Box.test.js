import React from 'react';
import { render } from '@testing-library/react';
import Box from './Box';

it('renders without crashing', function() {
	render(<Box key={1} height={100} width={100} backgroundColor="red" />);
});

it('matches snapshot', function() {
	const { asFragment } = render(<Box key={2} height={200} width={200} backgroundColor="yellow" />);
	expect(asFragment()).toMatchSnapshot();
});
