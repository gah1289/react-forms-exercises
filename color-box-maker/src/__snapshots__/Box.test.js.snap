// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class=" Box "
    style="background-color: yellow; width: 200px; height: 200px;"
  >
     
    <button
      class="delete-btn"
    >
      x
    </button>
  </div>
</DocumentFragment>
`;
